src/repos/UserRepo.ts

```diff
@@ -193,19 +193,116 @@ const incrementShoutouts = async (uuid: string): Promise<User> => {
 };
 
 /**
- * Increment number of bookmarks in user's numBookmarkedArticles
+ * Add article to a user's bookmarkedArticles
  */
-const incrementBookmarks = async (uuid: string): Promise<User> => {
+const bookmarkArticle = async (uuid: string, articleID: string): Promise<User> => {
   const user = await UserModel.findOne({ uuid });
 
   if (user) {
-    user.numBookmarkedArticles++;
+    const article = await ArticleRepo.getArticleByID(articleID);
+    const checkDuplicates = (prev: boolean, cur: Article) => prev || cur.id === articleID;
+
+    if (article && !user.bookmarkedArticles.reduce(checkDuplicates, false)) {
+      user.bookmarkedArticles.push(article);
+    }
+
+    user.save();
+  }
+
+  return user;
+};
+
+/**
+ * Add magazine to a user's bookmarkedMagazines
+ */
+const bookmarkMagazine = async (uuid: string, magazineID: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+
+  if (user) {
+    const magazine = await MagazineRepo.getMagazineByID(magazineID);
+    const checkDuplicates = (prev: boolean, cur: Magazine) => prev || cur.id === magazineID;
+
+    if (magazine && !user.bookmarkedMagazines.reduce(checkDuplicates, false)) {
+      user.bookmarkedMagazines.push(magazine);
+    }
+
+    user.save();
+  }
+
+  return user;
+};
+
+/**
+ * Add flyer to a user's bookmarkedFlyers
+ */
+const bookmarkFlyer = async (uuid: string, flyerID: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+
+  if (user) {
+    const flyer = await FlyerRepo.getFlyerByID(flyerID);
+    const checkDuplicates = (prev: boolean, cur: Flyer) => prev || cur.id === flyerID;
+
+    if (flyer && !user.bookmarkedFlyers.reduce(checkDuplicates, false)) {
+      user.bookmarkedFlyers.push(flyer);
+    }
+
     user.save();
   }
 
   return user;
 };
 
+/**
+ * Remove article from a user's bookmarkedArticles
+ */
+const unbookmarkArticle = async (uuid: string, articleID: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+  if (user) {
+    const articleSlugs = user.bookmarkedArticles.map((article) => article.id);
+    const articleIndex = articleSlugs.indexOf(articleID);
+
+    if (articleIndex === -1) return user;
+
+    user.followedOrganizations.splice(articleIndex, 1);
```

Comment:
Why are you doing this for `followedOrganizations` for bookmarking articles? Shouldn't you use `bookedmarkedArticles`?